nucleus/src/main/java/za/co/absa/subatomic/infrastructure/team/TeamAutomationHandler.java

```diff
@@ -64,21 +64,40 @@ public void createNewTeam(TeamEntity newTeamEntity) {
                 new TeamMemberId(newTeamEntity.getCreatedBy().getMemberId()),
                 teamSlackIdentity);
 
+        AtomistMemberBase atomistMemberBase = new AtomistMemberBase(
+                teamMemberEntity.getFirstName(),
+                teamMemberEntity.getDomainUsername(),
+                teamMemberSlackIdentity);
+
         TeamCreatedWithDetails newTeam = new TeamCreatedWithDetails(teamCreated,
-                new AtomistMemberBase(
-                        teamMemberEntity.getFirstName(),
-                        teamMemberEntity.getDomainUsername(),
-                        teamMemberSlackIdentity));
+                atomistMemberBase);
 
         ResponseEntity<String> response = restTemplate.postForEntity(
                 atomistConfigurationProperties.getTeamCreatedEventUrl(),
                 newTeam,
                 String.class);
 
         if (response.getStatusCode().is2xxSuccessful()) {
-            log.info("Atomist has ingested event successfully: {} -> {}",
+            log.info("Atomist has ingested teamCreatedEvent successfully: {} -> {}",
                     response.getHeaders(), response.getBody());
         }
+
+        // Raise event for teamSlackChannelCreated
```

Comment:
This section below looks fine, except that it is in the wrong place. You need to add a new function on the TeamAutomationHandler (something like `teamSlackChannelAssigned` or `teamSlackChannelCreated`). Then all this needs to go in that function, and needs to be called from here:
https://github.com/absa-subatomic/gluon/blob/384c7116dbacfcb90ca555053b75d95be3633f05/nucleus/src/main/java/za/co/absa/subatomic/application/team/TeamService.java#L142